### DISCUSSION TITLE
Not able to make AGH running on Raspberry Pi 4B with Raspian Bullseye 11 64 bits on it

### DISCUSSION THREAD

**Discussion by diegofrances** [AdguardTeam/AdGuardHome#6855] [Q&A]
Hello All,

I have installed Raspbian OS Desktop 64 bit (Debian 11 Bullseye) on a Raspberry Pi 4B. I have followed the guide explained at "https://github.com/AdguardTeam/AdGuardHome?tab=readme-ov-file#automated-install-linux-and-mac" but unfortunately I am not able to make it working.

When I send a request from the browser on the Raspbian, the AGH works as expected but when I point other machines to use the Raspberry Pi as the DNS server it does not work. I have taken a trace using Wireshark at client side and I can see the DNS queries reaching the Raspberry Pi but no action from the AGH is seen.

I have also noticed that when I run "netstat -atunp | grep 53", only tcp6 and udp6 are listening on port 53. Even I have disabled IPv6, I am not able to make it working on IPv4 so that's why I believe it is not working.

Can somebody please help me here?

Willing to share any log if need. I've ran out of ideas.

Thank you in advance.

Regards

**Comment by ghost** [comment]
G'day!

In the .yaml, could you please tell me which interface AdGuard Home is listening on by pasting the parameter for `bind_host`?

**Reply by diegofrances** [reply]
Hello,

I have reinstalled and instead of selecting all interfaces to listen on port 53, I have selected only the IPv4 one and now I can see the service listening on port 53 but unfortunately I am facing the same issue as I have described before, it only works on the local machine.

Please find the first lines of the yaml file:

http:
  pprof:
    port: 6060
    enabled: false
  address: 0.0.0.0:3000
  session_ttl: 720h
users:
  - name: username
    password: encrypted_password
auth_attempts: 5
block_auth_min: 15
http_proxy: ""
language: es
theme: dark
dns:
  bind_hosts:
    - 192.168.3.20
  port: 53

The above seen IP is the IP of the Raspberry Pi.

netstat -atunp | grep ":53 "
tcp        0      0 192.168.3.20:53         0.0.0.0:*               LISTEN      164237/AdGuardHome
udp        0      0 192.168.3.20:53         0.0.0.0:*                           164237/AdGuardHome

As you can see, the DNS config is as expected:
![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/dfb82011-e714-4f75-9d1e-ab2f5b3a9d0a)

When I try to test it, it fails:

![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/f757445c-8e77-4293-8663-c2f2da066dfa)

![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/055c42a6-da11-474f-befb-cd935fd4326a)

In opposition, when I ran the request from the Raspberry Pi, it works:

![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/1c846833-c59e-4cc0-91ca-9d7de0c342e1)

**Reply by ghost** [reply]
Are the queries coming through to the query log from that client?

**Reply by diegofrances** [reply]
Hello,

Unfortunately that's the issue that I am facing. Even if the IP of the device where the AdGuard is running is answering the DNS queries as per below image, the added block or any other rules are not being honoured.

![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/1d364103-ca2f-4d6b-831d-89e1b26fda69)

No entries at the AdGuard for the SandBox IP:
![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/a1ea5574-2e42-4505-ba50-f6fa4bf87c54)

SandBox's Config pointing to the IP where the AdGuard is running on:
![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/340b23a4-4b9f-4545-ade3-8aa2ef7e0ca5)

But as I said, any request coming from the Raspberry Pi itself is being analysed and filtered if need:
![imagen](https://github.com/AdguardTeam/AdGuardHome/assets/119085726/5fcc7b6a-b482-408f-a792-e1b5a1b5ef20)

Content of "dhcpcd.conf":
interface eth0
static ip_address=192.168.3.20
static routers=192.168.3.1
static domain_name_servers=192.168.3.20
noipv6
static domain_search=


Do I have to set the IP of the Raspberry out of the DHCP range of my Ubiquiti's Dream Router in order to work? Could that be the issue?

Thank you all for your collaboration.

**Reply by diegofrances** [reply]
Hi All,

I have been able to make it working but only for the devices sitting on the same VLAN/Subnet as the AdGuard. I have a Unifi Dream Router and even if I can exchange ICMP and DNS traffic between clients on different VLANs, the DNS requests coming from a different subnet/VLAN different from where the AdGuard is, is not being honoured.

I need to investigate further with UniFi Team.
